### ISSUE TITLE
[BUG] Neural Network models don't seem to get trained when using model.fit

### ISSUE THREAD

**Issue by cevheck** [sktime/sktime#5806]
**Describe the bug**
When using the ResNetClassifier (and the FCNClassifier), I found that my accuracy on my training set was very bad. After setting verbose=True I found out that, even though training appeared very good, the fitted model predicts almost random. See below a toy example.

from sktime.classification.deep_learning import ResNetClassifier, FCNClassifier
import numpy as np

**To Reproduce**
X_train = np.random.rand(10, 1000)
Y_train = np.array([np.sum(X_train[i,:]) > 500 for i in range(X_train.shape[0])])
model = ResNetClassifier(n_epochs=100, batch_size=32, activation="sigmoid", verbose=True)
model.fit(X_train, Y_train)
Y_pred_train = model.predict(X_train)
Y_pred_score = model.score(X_train, Y_train)

print(f"train_score = {sum(Y_pred_train == Y_train)/len(Y_train)}")
print(f'score_function score: {Y_pred_score}')

**Expected behavior**
verbose output at the last epoch to be equal to the real training accuracy / accuracy on the training set

**Output**
![image](https://github.com/sktime/sktime/assets/61511837/0dd1fdb6-a9cf-44f8-967e-15626db146a7)

System:
    python: 3.10.0 (default, Mar  3 2022, 09:58:08) [GCC 7.5.0]
   machine: Linux-5.15.0-91-generic-x86_64-with-glibc2.31

Python dependencies:
          pip: 23.3.1
       sktime: 0.25.0
      sklearn: 1.3.2
       skbase: 0.6.2
        numpy: 1.26.3
        scipy: 1.11.4
       pandas: 2.1.4
   matplotlib: 3.8.2
       joblib: 1.3.2
        numba: 0.58.1
  statsmodels: None
     pmdarima: None
statsforecast: None
      tsfresh: None
      tslearn: None
        torch: None
   tensorflow: None
tensorflow_probability: None

**Details
Already created a discussion
https://github.com/sktime/sktime/discussions/5804

**Comment by benHeid** [comment]
Hi @cevheck, 
thank you for reporting this. However, I do not think that this is in Bug. The reason of this behavior are the BatchNormalization Layers. These layers behave differently during training and inference. Thus, the result printed via the training differ compared to the results calculated afterwards. 

You might take a further look into: 
* https://keras.io/api/layers/normalization_layers/batch_normalization/
* https://github.com/keras-team/keras/issues/6977
* If you clone sktime you can test it locally also by commenting out all the BatchNormalisation Layers.
